src/main/java/net/sf/mzmine/modules/peaklistmethods/io/gnpsexport/GNPSExportTask.java

```diff
@@ -159,7 +176,7 @@ private void export(PeakList peakList, FileWriter writer, File curFile)
                 
                 String mass = Double.toString(row.getAverageMZ());
                 if (mass != null) writer.write("PEPMASS=" + mass + newLine);                                                                      
-                writer.write("CHARGE=1"+newLine);
+                writer.write("CHARGE="+row.getRowCharge()+"+"+newLine);
```

Comment:
row.getRowCharge() will return 0 if the charge is unknown or inconsistent among the peaks in the row. 
In such case, it might be better to avoid writing the CHARGE=0 line, no?